hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -675,7 +677,22 @@ protected Option<String> scheduleTableServiceInternal(String instantTime, Option
     return option;
   }
 
-  protected abstract HoodieTable<?, I, ?, T> createTable(HoodieWriteConfig config, StorageConfiguration<?> storageConf);
+  protected HoodieTable createTableAndValidate(HoodieWriteConfig config,
```

Comment:
method that centralizes table version validation. This method is called from all table services, to obtain `HoodieTable` before scheduling/planning.